Example for xxhash

I am trying to bring in xxhash with CPM (0.32) but I can't figure it out.

xxhash has an un-official cmake support, with a subfolder, that I got to work in the past with the SOURCE_SUBDIR magic value. Is *SOURCE_SUBDIR* supported with CPM ?

I am also trying to do it the DOWNLOAD_ONLY way, and then create a header only library.

```
if(xxhash_ADDED)
  add_library(xxhash INTERFACE)
  target_sources(xxhash INTERFACE ${xxhash_SOURCE_DIR}/xxhash.h ${xxhash_SOURCE_DIR}/xxh3.h)
  
  packageproject(...)
endif()

but I get lots of errors later on.

```
CMake Error in CMakeLists.txt:
  Target "xxhash" INTERFACE_SOURCES property contains path:

    "/Users/benjamin.sergeant/projects/xyz/build/_deps/xxhash-src/xxh3.h"

  which is prefixed in the build directory.
```

--- a/examples/xxHash/CMakeLists.txt
+++ b/examples/xxHash/CMakeLists.txt
@@ -7,12 +7,10 @@ project(CPMxxHashExample)
 include(../../cmake/CPM.cmake)
 
 CPMAddPackage(
-    GITHUB_REPOSITORY Cyan4973/xxHash
-    GIT_TAG v0.8.2
-    OPTIONS 
-        "XXHASH_BUILD_ENABLE_INLINE_API OFF"
-        "XXHASH_BUILD_XXHSUM OFF"
-    SOURCE_SUBDIR cmake_unofficial
+  GITHUB_REPOSITORY Cyan4973/xxHash
+  GIT_TAG v0.8.2
+  OPTIONS "XXHASH_BUILD_ENABLE_INLINE_API OFF" "XXHASH_BUILD_XXHSUM OFF"
+  SOURCE_SUBDIR cmake_unofficial
 )
 
 # ---- Executable ----

--- a/examples/xxHash/CMakeLists.txt
+++ b/examples/xxHash/CMakeLists.txt
@@ -1,0 +1,22 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(CPMxxHashExample)
+
+# ---- Dependencies ----
+
+include(../../cmake/CPM.cmake)
+
+CPMAddPackage(
+    GITHUB_REPOSITORY Cyan4973/xxHash
+    GIT_TAG v0.8.2
+    OPTIONS 
+        "XXHASH_BUILD_ENABLE_INLINE_API OFF"
+        "XXHASH_BUILD_XXHSUM OFF"
+    SOURCE_SUBDIR cmake_unofficial
+)
+
+# ---- Executable ----
+
+add_executable(CPMxxHashExample main.cpp)
+target_compile_features(CPMxxHashExample PRIVATE cxx_std_17)
+target_link_libraries(CPMxxHashExample xxHash::xxHash)

--- a/examples/xxHash/CMakeLists.txt
+++ b/examples/xxHash/CMakeLists.txt
@@ -19,4 +19,4 @@ CPMAddPackage(
 
 add_executable(CPMxxHashExample main.cpp)
 target_compile_features(CPMxxHashExample PRIVATE cxx_std_17)
-target_link_libraries(CPMxxHashExample xxHash::xxHash)
+target_link_libraries(CPMxxHashExample xxHash::xxhash)

--- a/examples/xxHash/main.cpp
+++ b/examples/xxHash/main.cpp
@@ -1,0 +1,13 @@
+#include <iostream>
+#include <xxh3.h>
+
+int main() {
+
+  std::string example = "Hello World!";
+  XXH64_hash_t hash = XXH3_64bits(example.data(), example.size());
+
+  std::cout << "Hash: " << hash << std::endl;
+  
+  return 0;
+
+}

--- a/examples/xxHash/main.cpp
+++ b/examples/xxHash/main.cpp
@@ -1,13 +1,12 @@
-#include <iostream>
 #include <xxh3.h>
 
-int main() {
+#include <iostream>
 
+int main() {
   std::string example = "Hello World!";
   XXH64_hash_t hash = XXH3_64bits(example.data(), example.size());
 
   std::cout << "Hash: " << hash << std::endl;
-  
-  return 0;
 
+  return 0;
 }

